Build failure on -tags mobile when updating to Fedora 35

<!-- Please search for open issues that relate to the same problem before opening a new one. -->

### Describe the bug:
I recently updated to Fedora 35 and it seems that I no longer can run the mobile simulator using `-tags mobile`. I am now getting the following error:
```go
# fyne.io/fyne/v2/internal/driver/mobile/app
x11.c: In function ‘new_window’:
x11.c:40:9: error: unknown type name ‘XVisualInfo’; did you mean ‘VisualID’?
   40 |         XVisualInfo visTemplate;
      |         ^~~~~~~~~~~
      |         VisualID
x11.c:41:20: error: request for member ‘visualid’ in something not a structure or union
   41 |         visTemplate.visualid = vid;
      |                    ^
x11.c:43:9: error: unknown type name ‘XVisualInfo’; did you mean ‘VisualID’?
   43 |         XVisualInfo *visInfo = XGetVisualInfo(x_dpy, VisualIDMask, &visTemplate, &num_visuals);
      |         ^~~~~~~~~~~
      |         VisualID
x11.c:43:32: warning: implicit declaration of function ‘XGetVisualInfo’ [-Wimplicit-function-declaration]
   43 |         XVisualInfo *visInfo = XGetVisualInfo(x_dpy, VisualIDMask, &visTemplate, &num_visuals);
      |                                ^~~~~~~~~~~~~~
x11.c:43:54: error: ‘VisualIDMask’ undeclared (first use in this function); did you mean ‘VisualID’?
   43 |         XVisualInfo *visInfo = XGetVisualInfo(x_dpy, VisualIDMask, &visTemplate, &num_visuals);
      |                                                      ^~~~~~~~~~~~
      |                                                      VisualID
x11.c:43:54: note: each undeclared identifier is reported only once for each function it appears in
x11.c:52:61: error: request for member ‘visual’ in something not a structure or union
   52 |         attr.colormap = XCreateColormap(x_dpy, root, visInfo->visual, AllocNone);
      |                                                             ^~
x11.c:61:52: error: request for member ‘depth’ in something not a structure or union
   61 |                 x_dpy, root, 0, 0, w, h, 0, visInfo->depth, InputOutput,
      |                                                    ^~
x11.c:62:24: error: request for member ‘visual’ in something not a structure or union
   62 |                 visInfo->visual, CWColormap | CWEventMask, &attr);
      |                        ^~
x11.c:65:9: error: unknown type name ‘XSizeHints’
   65 |         XSizeHints sizehints;
      |         ^~~~~~~~~~
x11.c:66:18: error: request for member ‘width’ in something not a structure or union
   66 |         sizehints.width  = w;
      |                  ^
x11.c:67:18: error: request for member ‘height’ in something not a structure or union
   67 |         sizehints.height = h;
      |                  ^
x11.c:68:18: error: request for member ‘flags’ in something not a structure or union
   68 |         sizehints.flags = USSize;
      |                  ^
x11.c:68:27: error: ‘USSize’ undeclared (first use in this function)
   68 |         sizehints.flags = USSize;
      |                           ^~~~~~
x11.c:69:9: warning: implicit declaration of function ‘XSetNormalHints’ [-Wimplicit-function-declaration]
   69 |         XSetNormalHints(x_dpy, win, &sizehints);
      |         ^~~~~~~~~~~~~~~
x11.c:70:9: warning: implicit declaration of function ‘XSetStandardProperties’ [-Wimplicit-function-declaration]
   70 |         XSetStandardProperties(x_dpy, win, "App", "App", None, (char **)NULL, 0, &sizehints);
      |         ^~~~~~~~~~~~~~~~~~~~~~

```


### To Reproduce:
Steps to reproduce the behaviour:
1. Compile `cmd/hello` using `-tags mobile`.

### Example code:
<!-- If applicable, add a short code snippet to help explain and simplify reproduction of the problem. 
Please refrain from adding screenshots of code, links to other projects or very long code examples. -->
<!-- Please write the code inside a code block with go syntax highlighting enabled, like this:
```go
Write your code here.
```
-->
Compile any Fyne app, for example `cmd/hello` or `cmd/fyne_demo`.


### Device (please complete the following information):
 - **OS:** <!-- [e.g. Linux, MacOS or iOS] --> Fedora Linux 35
 - **Version:** <!-- [e.g. 5.10.2, 10.13 High Sierra or 14.2] --> 5.14.15
 - **Go version:** <!-- [e.g. 1.14.3] --> 1.16.8
 - **Fyne version:** <!-- [e.g. 2.1.0 or git SHA] --> 2.1.1 and on 2.1.x branch.

--- a/internal/driver/mobile/app/x11.c
+++ b/internal/driver/mobile/app/x11.c
@@ -8,6 +8,7 @@
 #include <EGL/egl.h>
 #include <GLES2/gl2.h>
 #include <X11/Xlib.h>
+#include <X11/Xutil.h>
 #include <stdio.h>
 #include <stdlib.h>
 

